infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -135,107 +142,103 @@ public void checkpointState() {
         }
     }
 
-    @Override
-    public synchronized void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        switch (msg.getMsgType()) {
-            case TOKEN_REQ: {
-                TokenRequest req = ((CorfuPayloadMsg<TokenRequest>) msg).getPayload();
-                if (req.getNumTokens() == 0) {
-                    long max = 0L;
-                    boolean hit = false;
-                    ImmutableMap.Builder<UUID, Long> streamsLastIssued = ImmutableMap.builder();
+    /**
+     * Service an incoming token request.
+     */
+    @ServerHandler(type=CorfuMsgType.TOKEN_REQ)
+    public synchronized void tokenRequest(CorfuPayloadMsg<TokenRequest> msg,
+                                          ChannelHandlerContext ctx, IServerRouter r) {
+        TokenRequest req = msg.getPayload();
+        if (req.getNumTokens() == 0) {
+            long max = 0L;
+            boolean hit = false;
+            ImmutableMap.Builder<UUID, Long> streamsLastIssued = ImmutableMap.builder();
+            for (UUID id : req.getStreams()) {
```

Comment:
I found it really difficult to understand this code with the similar names lastIssued, streamsLastIssued, lastLocalOffstMap, lastIssuedMap.

1. change names to more meaningful ones
2. comment this code, it is central to the token/stream reservation utility